tests/suites/test_suite_ssl.function

```diff
@@ -177,22 +181,29 @@ int mbedtls_test_message_queue_setup( mbedtls_test_message_queue *queue,
 
 void mbedtls_test_message_queue_free( mbedtls_test_message_queue *queue )
 {
+    if( queue == NULL )
+        return;
+
     if( queue->messages != NULL )
         mbedtls_free( queue->messages );
 
     memset( queue, 0, sizeof( *queue ) );
 }
 
 /*
- * Push length information onto the message metadata queue. This will become
- * the last element to leave it (fifo).
+ * Push message length information onto the message metadata queue.
+ * This will become the last element to leave it (fifo).
+ *
+ * \retval  MBEDTLS_TEST_ERROR_ARG_NULL, if the queue is null.
+ * \retval  MBEDTLS_TEST_ERROR_QUEUE_FULL, if the queue is full.
+ * \retval  0, if the push was succesfull.
```

Comment:
This should be added in "Add a message metadata queue in ssl tests", not "Add a message-based socket mock connection to the ssl tests"